File: backend/src/api/dtos/fixture.py
"""Pydantic response DTOs for the /api/fixtures routes."""

from datetime import datetime

from pydantic import BaseModel

from src.domain.match.fixture import Fixture


class FixtureResponse(BaseModel):
    id: int
    home_team_id: str
    away_team_id: str
    status: str
    group: str
    home_score: int | None = None
    away_score: int | None = None
    kickoff_at: datetime | None = None
    minute: int | None = None
    note: str | None = None
    venue_name: str | None = None
    venue_city: str | None = None
    stage_name: str | None = None
    round_name: str | None = None

    @classmethod
    def from_domain(cls, fixture: Fixture) -> "FixtureResponse":
        return cls(
            id=fixture.id,
            home_team_id=fixture.home_team_id,
            away_team_id=fixture.away_team_id,
            status=fixture.status.value,
            group=fixture.group,
            home_score=fixture.home_score,
            away_score=fixture.away_score,
            kickoff_at=fixture.kickoff_at,
            minute=fixture.minute,
            note=fixture.note,
            venue_name=fixture.venue_name,
            venue_city=fixture.venue_city,
            stage_name=fixture.stage_name,
            round_name=fixture.round_name,
        )

File: packages/core/src/domain/match/fixture.ts
export type FixtureStatus = "live" | "finished" | "upcoming";

export interface Fixture {
  id: number;
  home_team_id: string;
  away_team_id: string;
  status: FixtureStatus;
  group: string;
  home_score?: number;
  away_score?: number;
  /** Penalty-shootout score (converted penalties per side). Present only when
   * the knockout was decided on penalties; the winner is the higher of the two. */
  home_pen_score?: number;
  away_pen_score?: number;
  date?: string;
  minute?: number;
  note?: string;
  /** Stadium name (e.g. "Lusail Stadium"). */
  venue_name?: string;
  /** Host city of the venue (e.g. "Lusail"). */
  venue_city?: string;
  /** Sportmonks stage label: "Group Stage", "Round of 16", "Quarter-finals",
   * "Semi-finals", "3rd Place Final", "Final". */
  stage_name?: string;
  /** Matchday number within a stage (e.g. "1", "2", "3" for the group stage).
   * Null/undefined for knockout fixtures. */
  round_name?: string;
}
